Makefile

```diff
@@ -6,7 +6,10 @@ venv:
 install: venv
 	. venv/bin/activate; pip install .
 
-test-install: install
+pre-test-install:
+	@`python --version || python3 --version` || (echo "Python is not installed, please install Python 2 or Python 3"; exit 1)
```

Comment:
don't think this is necessary because if only python 3 is installed it will be aliased to just python so dont need to check that specifically.

Also we probably want to just add this line to the venv target or something since none of those targets will succeed if python isnt installed.